src/player.cpp

```diff
@@ -7350,7 +7354,25 @@ void player::hardcoded_effects(effect &it)
                 }
             }
         }
+    } else if (id == "grabbed"){
+        int intensity = 0;
+        for (int i = -1; i <= 1; i++) {
+            for (int j = -1; j <= 1; j++) {
+                tripoint dest( posx() + i, posy() + j, posz() );
+                if (g->mon_at(dest) != -1){
+                    intensity ++;
+                }
+            }
+        }
+        if (intensity > 0){
+            if (get_effect_int("grabbed") > intensity * 2){
```

Comment:
The point of the formula is to cap it at 2 * number of zeds. If zed number is more than nothing, than the grab intensity is kept as is. If there is at least 1 zed around, it'll check the grab intensity, and if it is more than 2 * zed number, it'll replace the effect with one that has it's intensity at 2 * zed number.